### PR TITLE
Upgrade Colony Modal

### PR DESCRIPTION
**Pull request by chinins**
## Description

This issue refactors the existing colony upgrade modal to incorporate new features and a bit of a UI refresh.

**New stuff** ✨

- add form components
- add annotation
- add contract versions

**Changes** 🏗

- change the logic & appearance similar to other dialogs

**Deletions** ⚰️

- none

## TODO

- none

Resolves DEV-72

Screenshot 2021-01-07 at 11 57 29
Screenshot 2021-01-07 at 12 59 58

### CODE REVIEW

**Comment by rdig** [comment]
Hey @chinins I think @alicjakujawa already implemented this modal in #2338 

Please check if that's the case

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Looking fresh :ok_hand: I just noticed something small and it's that the labels and version numbers here:

FireShot Capture 071 - Colony - localhost

Should be in `700`, like in the design:
FireShot Capture 076 - Colony Simplified – Figma - www figma com

File: src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx
Comment (rdig) [inline src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx:101]: Please change this so that the version lines span the whole width of the dialog (basically remove the right and left margins)

File: src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx
Comment (rdig) [inline src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx:93]: Please reduce the font size to `13px` so that it's aligned with the figma styles

Screenshot from 2021-01-12 11-29-01

File: src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx
Comment (rdig) [inline src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx:94]: Please reduce the size of the version number (as mentioned above), but also have this number have a font weight of `700`.

Screenshot from 2021-01-12 11-29-12

Same goes for the network version

File: src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx
Comment (rdig) [inline src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx:126]: Please reduce the top padding of this dialog section so that the error has less empty space before it

Screenshot from 2021-01-12 11-30-04

**Review by rdig — CHANGES_REQUESTED** [review]
Yeah, this is mostly good, there's just a couple of styles changes I want you to make.

I've noted them below

**Comment by chinins** [comment]
> Looking fresh 👌 I just noticed something small and it's that the labels and version numbers here:
> 
> FireShot Capture 071 - Colony - localhost
> 
> Should be in `700`, like in the design:
> FireShot Capture 076 - Colony Simplified – Figma - www figma com

I asked Raul and he said this should be the format of the contract.

File: src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx
Comment (chinins) [reply src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx:126]: The same padding is used in all other dialogs. Should we then reduce the top padding in all other dialogs as well for consistency purpose?

**Comment by ArmandoGraterol** [comment]
> > Looking fresh ok_hand I just noticed something small and it's that the labels and version numbers here:
> > FireShot Capture 071 - Colony - localhost
> > Should be in `700`, like in the design:
> > FireShot Capture 076 - Colony Simplified – Figma - www figma com
> 
> I asked Raul and he said that my just `5` was an appropriate version

I understand, but I wasn't referring to that. What I mean is that the `Current Version`, `New Version`, and both `5` should be in `font-weight: 700` ... No matter, Raul also mentioned it, and I saw that you marked it as resolved so all good now :)

I'll wait until the changes are uploaded!

File: src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx
Comment (rdig) [inline src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx:93]: So the "Current Version" copy needs to have a normal font weight (eg: 400), while the actual version number itself should be bold (font weight 700)

Screenshot from 2021-01-12 18-18-02

Yours are both in weight 700

File: src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx
Comment (rdig) [inline src/modules/dashboard/components/NetworkContractUpgradeDialog/NetworkContractUpgradeDialogForm.tsx:92]: As I've said previous,ly please reduce the left/right padding, so that the version lines span the whole width of the modal.

Yours:
Screenshot from 2021-01-12 18-17-05

How I want this to look:
Screenshot from 2021-01-12 18-16-43

**Review by rdig — APPROVED** [review]
There are still two more chances to be done -- either you missed them in the previous review, or just forgot about them (it happens)

I've marked them below, but I'll pre-approve this so that after you fix them you can merge.

**Pull request merged**
